libraries/standard/json/src/json.c

```diff
@@ -0,0 +1,1014 @@
+ */
+static bool skipSpaceAndComma( const char * buf,
+                               size_t * start,
+                               size_t max )
+{
+    bool ret = false;
+    size_t i;
+
+    skipSpace( buf, start, max );
+    i = *start;
+
+    if( ( i < max ) && ( buf[ i ] == ',' ) )
+    {
+        i++;
+        skipSpace( buf, &i, max );
+
+        if( ( i < max ) && ( buf[ i ] != '}' ) && ( buf[ i ] != ']' ) )
+        {
+            ret = true;
+            *start = i;
+        }
+    }
+
+    return ret;
+}
+
+/**
+ * @brief Advance buffer index beyond the scalar values of an array.
+ *
+ * @param[in] buf  The buffer to parse.
+ * @param[in,out] start  The index at which to begin.
+ * @param[in] max  The size of the buffer.
+ *
+ * @return true if a scalar value was present;
+ * false otherwise.
+ *
+ * @note Stops advance if a value is an object or array.
+ */
+static bool skipArrayScalars( const char * buf,
+                              size_t * start,
+                              size_t max )
+{
+    bool ret = false;
+    size_t i = *start;
+
+    while( i < max )
+    {
+        if( skipAnyScalar( buf, &i, max ) != true )
+        {
+            break;
+        }
+
+        ret = true;
+
+        if( skipSpaceAndComma( buf, &i, max ) != true )
+        {
+            break;
+        }
+    }
+
+    *start = i;
+    return ret;
+}
+
+/**
+ * @brief Advance buffer index beyond the scalar key-value pairs
+ * of an object.
+ *
+ * In JSON, objects consist of comma-separated key-value pairs.
+ * A key is always a string (a scalar) while a value may be a
+ * scalar, an object, or an array.
+ *
+ * @param[in] buf  The buffer to parse.
+ * @param[in,out] start  The index at which to begin.
+ * @param[in] max  The size of the buffer.
+ *
+ * @return true if a key was present;
+ * false otherwise.
+ *
+ * @note Stops advance if a value is an object or array.
+ */
+static bool skipObjectScalars( const char * buf,
+                               size_t * start,
+                               size_t max )
+{
+    bool ret = false;
+    size_t i = *start;
+
+    while( i < max )
+    {
+        if( skipString( buf, &i, max ) != true )
+        {
+            break;
+        }
+
+        skipSpace( buf, &i, max );
+
+        if( ( i < max ) && ( buf[ i ] != ':' ) )
+        {
+            break;
+        }
+
+        i++;
+        ret = true;
+        skipSpace( buf, &i, max );
+
+        if( skipAnyScalar( buf, &i, max ) != true )
+        {
+            break;
+        }
+
+        if( skipSpaceAndComma( buf, &i, max ) != true )
+        {
+            break;
+        }
+    }
+
+    *start = i;
+    return ret;
+}
+
+/**
+ * @brief Advance buffer index beyond one or more scalars.
+ *
+ * @param[in] buf  The buffer to parse.
+ * @param[in,out] start  The index at which to begin.
+ * @param[in] max  The size of the buffer.
+ *
+ * @return true if a scalar was present;
+ * false otherwise.
+ */
+static bool skipScalars( const char * buf,
```

Comment:
return value here doesn't seem to be used. other functions whose return value would be ignored (e.g. skipExponent) seem to return void. would be nice to either always return bool or always use void when the return value won't be used.

Reply:
changed to use void for 3 functions